### ISSUE TITLE
Ten Days Rule correction coefficient

### ISSUE THREAD

**Issue by yegor256** [zerocracy/farm#558]
How about we add a correction to the Ten Days Rule? The higher the reputation of a user, the bigger the amount of time we can wait for a job completion. The lower the reputation, the faster the user has to act. The motivation here is obvious: new low-reputation users put projects on hold, since they can't complete in time even the simplest tasks. We should give them 3-4 days and eject as soon as possible. On the other hand, high-reputation users in most cases have good reasons for holding jobs longer. Let's give them 16-20 days or so. WDYT?

**Comment by amihaiemil** [comment]
@yegor256 I think we can wait more days for higher-reputation contributors, but we should leave it the same for new commers. 3-4 days is way to little time, taking into account that 4 people are involved: the ticket's author, the resolver, the CR and the ARCH. 

Also, let's not forget that the model should be distributed, so this also means flexibility. If I am forced to really hurry and put off other things in order to close a task quickly, then we don't really have flexibility anymore. 

This could also be configurable. The PO should have some sort of project dashboard on 0crat.com where they could set these deadlines.

**Comment by yegor256** [comment]
@amihaiemil well, I think it's better to "fail fast" with newcomers and give them feedback earlier. They will either understand the importance of doing everything quickly (by losing points) or will quit. But the project won't suffer. It's sort of a faster training.

On the other hand, totally fresh/zero-reputation newcomers are only joining sandbox projects, which don't really care about the speed of delivery. So, maybe you are right.

**Comment by amihaiemil** [comment]
@yegor256 Yes, speed is the most important part. After all, the flexibility depends on the project.  The developers will have to adjust anyway. 

Still, wouldn't it be better to let each project configure this timeframe? 3-4 days is good from the "fail-fast" perspective you are mentioning, but maybe some projects don't like this idea.

**Comment by yegor256** [comment]
@amihaiemil I think that we should make our software as less configurable as possible. Becuase if it will have many configuration options, it will not translate any particular philosophy/concept to the projects. It will be just a piece of software. On the other hand, if it works only one specific way, people will buy/take not only its functionality, but also the concept behind it.

**Comment by amihaiemil** [comment]
Right, that's also a point... well, then I say we can give it a shot :)

**Comment by driver733** [comment]
@yegor256 I believe that we should give devs more time to complete the tasks at the same time they get their agenda increased. For instance, when your reputation reaches 512, you can have 8 tasks at the same time instead of 3, when you reputation was lower. 
I suggest the following:

```
Reputation (points) - time for each task (days)
<256 (3 tasks) - 7
<512 (3 tasks) - 10
<2048 (8 tasks) - 20
<4096 (16 tasks) - 30
>4096 (24 tasks) - 40

```
I also agree that the system should not be configurable from outside.

**Comment by yegor256** [comment]
@driver733 @amihaiemil maybe it's better to keep 10 days as the lower margin and increase it when the reputation grows: 

  * +0 days if less than 512 points, 
  * +2 days if less than 2048, 
  * +4 days if less than 4096, 
  * +8 days otherwise.

WDYT?

**Comment by yegor256** [comment]
@g4s8 @carlosmiranda what is you opinion here?

**Comment by amihaiemil** [comment]
@yegor256 Sure 18 days (best case proposed) is more than enough, But the best and least complicated mechanism would be to just count the person for which the task *as a whole* waits. 0crat should:

1) the assined DEV up until he/she made a PR (stop the count while there is a PR open)
2) start counting the CR when the PR is assigned 
3) stop counting the CR and start counting the ARCH when it's his/her turn
4) stop counting the ARCH and start counting the QA.

This way, I think we can even keep the 10 days rule. 10 days is more than enough for a single person -- trouble is that now everyone is counted at the same time, and they wait for each other a lot. 

It seems complex, but I think it would be easily implemented if 0crat could treat a task as a whole: Issue + opened PR(s). So far, as I know, 0crat treats each opened Issue/PR as a completely separate job, and they are not. 

I also think 0crat should do this because it advertises as AI project manager (I understood it shouldn't be AI in the commands/chatting part, but this is part of its operation, of its service). 

To summarize: 10 days is enough, we just need to count each person when the task as a whole is in their hands.

**Comment by yegor256** [comment]
@amihaiemil we can't pay attention to "opened PRs". It something Zerocrat should not care about. It's your (the performer of the ticket) business. You may close a job even without a PR or may have 5 of them and still never complete the job. So, I strongly believe we should stay away from any analysis of a job internals.

**Comment by g4s8** [comment]
@yegor256 I agree, but maybe it should be related to user's agenda , not reputation? He/she may have many REV tasks assigned with low reputation

**Comment by amihaiemil** [comment]
@yegor256 I see... ok, then let's go with your proposal. @g4s8 has a point too: the number of assigned task is also important :)

**Comment by yegor256** [comment]
@g4s8 @amihaiemil well, a busy agenda doesn't mean that we should give that person more time. Bigger time lag means higher trust to the person. We give him/her 18 days instead of 10 because we believe that he/she will use them effectively. We believe that the user needs that time because the job requires so. Not because the user is just busy.

**Comment by carlosmiranda** [comment]
@yegor256 Generally speaking I agree with your proposal.

As for the points raised by @g4s8 and @amihaiemil regarding number of tasks in agenda, I think that should be a separate feature. We want folks to turn over a small amount of tasks quickly instead of waiting on a large amount of tasks, so extending the deadline for multiple tasks is not efficient. See my proposal in #621.

**Comment by 0crat** [comment]
Job [#558](https://github.com/zerocracy/farm/issues/558) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10624](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#558](https://github.com/zerocracy/farm/issues/558) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/672143674). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by g4s8** [comment]
@yegor256 what is the final decision here? This https://github.com/zerocracy/farm/issues/558#issuecomment-367314678 ?

**Comment by g4s8** [comment]
@0crat waiting for answer https://github.com/zerocracy/farm/issues/558#issuecomment-373746674

**Comment by 0crat** [comment]
> @0crat waiting for answer https://github.com/zerocracy/farm/issues/558#issuecomment-373746674 ([here](https://github.com/zerocracy/farm/issues/558#issuecomment-373932400))

@g4s8 The impediment for [#558](https://github.com/zerocracy/farm/issues/558) was registered successfully by @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by yegor256** [comment]
@g4s8 yes, let's implement the schedule I proposed

**Comment by yegor256** [comment]
@g4s8 thanks!

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/558#issuecomment-378213804))

@g4s8 This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#558](https://github.com/zerocracy/farm/issues/558) is [in scope](http://www.zerocracy.com/policy.html#14) for 1mon
  * The role is `DEV`
  * The job is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8) for [26days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @g4s8[/z](https://www.0crat.com/u/g4s8) will get 50% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job has an [impediment](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @yegor256[/z](https://www.0crat.com/u/yegor256): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%23558%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/558#issuecomment-378604476))

@g4s8 This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#558](https://github.com/zerocracy/farm/issues/558) is [in scope](http://www.zerocracy.com/policy.html#14) for 1mon
  * The role is `DEV`
  * The job is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8) for [27days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @g4s8[/z](https://www.0crat.com/u/g4s8) will get 50% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job has an [impediment](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @yegor256[/z](https://www.0crat.com/u/yegor256): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%23558%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@yegor256 remove this job from WBS please, 0crat didn't do this automatically

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @g4s8[/z](https://www.0crat.com/u/g4s8), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/558#issuecomment-379691469))

@yegor256 The job [#558](https://github.com/zerocracy/farm/issues/558) is now out of scope

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/zerocracy/farm/issues/558#issuecomment-365994618) by indicating an addressee in the beginning.

**Comment by amihaiemil** [comment]
@ypshenychka done :)

**Comment by ypshenychka** [comment]
@amihaiemil Thanks!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +30 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
